samples/android/Continote/app/src/main/java/com/firebasecontinue/sample/continote/EditNoteActivity.java

```diff
@@ -0,0 +1,134 @@
+        if (mDatabaseKey == null) {
+            // We must have a database key to be on this screen.
+            finish();
+            return;
+        }
+
+        setContentView(R.layout.activity_edit_note);
+
+        // Gather the UI elements for this Activity for future manipulation.
+        mContinueWritingElsewhereButton =
+                (Button) findViewById(R.id.continueWritingElsewhereButton);
+    }
+
+    @Override
+    protected void handleUserSignedIn(@NonNull FirebaseUser user) {
+        super.handleUserSignedIn(user);
+
+        if (user == null) {
+            // This should never happen, but just in case.
+            throw new AssertionError("user must be non-null");
+        }
+
+        // TODO
+    }
+
+    @Override
+    protected void handleUserSignedOut() {
+        super.handleUserSignedOut();
+
+        // The user must be signed in to view this screen, so navigate away from it if the user is
+        // signed out.
+        finish();
+    }
+
+    /**
+     * Handles when the user taps the continueWritingElsewhereButton.
+     *
+     * @param v The View that called this triggered this handler.
+     *          This should only be the continueWritingElsewhereButton itself.
+     */
+    public void handleContinueWritingElsewhereButtonTapped(@Nullable View v) {
+        if (!currentUserIsSignedIn() || mDatabaseKey == null) {
+            // This should never happen, but just in case.
+            throw new AssertionError("User must be signed in and database key must be non-null");
+        }
+
+        // The current user is signed in, so allow them to easily continue writing this Note
+        // within Chrome via the Continote Chrome extension and Continote web app.
+
+        // TODO: Save the Note first so that when the user opens it elsewhere they see the latest
+        // values for the Note (that they presumably wrote here in this app).
+
+        // We use what's known as "activity-scoped listeners" here to react to the broadcast
+        // Task either succeeding or failing upon its completion.
+        // The reason we use activity-scoped listeners here is that they are automatically
+        // removed during the onStop method of the Activity so that they are not called when
+        // the Activity is no longer visible. For more details about activity-scoped listeners
+        // (and Task listeners in general), see:
+        // https://developers.google.com/android/guides/tasks#activity-scoped_listeners
+        FirebaseContinue.broadcastActivityToContinue(
+                getString(R.string.continote_url_to_edit_note_with_key, mDatabaseKey),
+                getString(R.string.app_name_for_firebase_continue)
+        ).addOnSuccessListener(this, new OnSuccessListener<Void>() {
+            @Override
+            public void onSuccess(Void result) {
+                showSnackbar(R.string.broadcast_to_continue_writing_note_in_chrome_successful);
```

Comment:
perhaps use a Toast instead of the snackbar? I've found that iOS tends to use snackbar in a lot of situations where Android uses Toast. That being said, Android does also have snackbar, but food for thought.